src/components/VideoPlayerPreview/index.js

```diff
@@ -0,0 +1,91 @@
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import {View} from 'react-native';
+import * as Expensicons from '@components/Icon/Expensicons';
+import VideoPlayer from '@components/VideoPlayer';
+import IconButton from '@components/VideoPlayer/IconButton';
+import {usePlaybackContext} from '@components/VideoPlayerContexts/PlaybackContext';
+import useThumbnailDimensions from '@hooks/useThumbnailDimensions';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import styles from '@styles/styles';
+import VideoPlayerThumbnail from './VideoPlayerThumbnail';
+
+const propTypes = {
+    videoUrl: PropTypes.string.isRequired,
+
+    // eslint-disable-next-line react/forbid-prop-types
+    videoDimensions: PropTypes.object,
+
+    thumbnailUrl: PropTypes.string,
+
+    fileName: PropTypes.string.isRequired,
+
+    showModal: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    videoDimensions: {width: 1900, height: 1400},
+    thumbnailUrl: null,
+};
+
+function VideoPlayerPreview({videoUrl, thumbnailUrl, fileName, videoDimensions, showModal}) {
+    const {currentlyPlayingURL, updateCurrentlyPlayingURL} = usePlaybackContext();
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const [isThumbnail, setIsThumbnail] = useState(true);
+    const [measuredDimenstions, setMeasuredDimenstions] = useState(videoDimensions);
+    const {thumbnailDimensionsStyles} = useThumbnailDimensions(measuredDimenstions.width, measuredDimenstions.height);
+
+    const onVideoLoaded = (e) => {
+        setMeasuredDimenstions({width: e.srcElement.videoWidth, height: e.srcElement.videoHeight});
+    };
+
+    const handleOnPress = () => {
+        updateCurrentlyPlayingURL(videoUrl);
+        if (isSmallScreenWidth) {
+            showModal();
+        }
+    };
+
+    useEffect(() => {
+        if (videoUrl !== currentlyPlayingURL) {
+            return;
+        }
+        setIsThumbnail(false);
+    }, [currentlyPlayingURL, updateCurrentlyPlayingURL, videoUrl]);
+
+    return (
+        <View style={[styles.webViewStyles.tagStyles.video, thumbnailDimensionsStyles]}>
+            {isSmallScreenWidth || isThumbnail ? (
+                <VideoPlayerThumbnail
+                    thumbnailUrl={thumbnailUrl}
+                    onPress={handleOnPress}
+                    accessibilityLabel={fileName}
+                />
+            ) : (
+                <>
+                    <VideoPlayer
+                        url={videoUrl}
+                        videoPlayerStyles={{borderRadius: 10}}
```

Comment:
Is the `10` value correct? As we usually use values which are multiple of 4?

I don't have access to the design-doc yet, so just asking.

Reply:
Right, I checked the figma file and we should use 8px border radius. Good catch, thanks

Reply:
Also, we don't use `videoPlayerStyles` anymore, so I will delete it